### PR TITLE
docs: made the setting_status page use the latest ready_t feature

### PR DESCRIPTION
**Pull request by Jaskowicz1**
This PR changes the setting_status page to use the feature added from #1038. I've also made the page move away from using the cache.

## Documentation change checklist

- [x] My documentation changes follow the [docs style guide](https://dpp.dev/docs-standards.html) and any code examples follow the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR (via running `doxygen`, and testing examples).
- [x] I have not moved any existing pages or changed any existing URLs without strong justification as to why.
- [x] I have not generated content using AI or a desktop utility such as grammarly.

### CODE REVIEW

File: docpages/example_programs/misc/setting_status.md [hunk 1]
```diff
@@ -2,8 +2,6 @@
 
 A bot status is pretty cool, and it'd be cooler if you knew how to do it! This tutorial will cover how to set the bot status to say `Playing games!`, as well as covering how to set the status to the amount of guilds every two minutes.
 
-\note dpp::get_guild_cache requires the bot to have the guild cache enabled, if your bot has this disabled then you can't use that. Instead, you should look to use dpp::cluster::current_application_get and get the `approximate_guild_count` from dpp::application in the callback.
```
Comment (braindigitalis) [inline docpages/example_programs/misc/setting_status.md:5]: why is this removed? it is a valid and important note.

**Review by braindigitalis — CHANGES_REQUESTED** [review]
see note

File: docpages/example_programs/misc/setting_status.md (hunk 1 above)
Comment (Jaskowicz1) [reply docpages/example_programs/misc/setting_status.md:5]: Because the code now uses that function, I made it no longer rely on the cache.

Comment (Jaskowicz1) [reply docpages/example_programs/misc/setting_status.md:5]: True, I'll undo that part then.

**Review by braindigitalis — APPROVED** [review]
